src/IMangrove.sol

```diff
@@ -1,27 +1,52 @@
 // SPDX-License-Identifier: Unlicense
 // This file was manually adapted from a file generated by abi-to-sol. It must
-// be kept up-to-date with the actual Mangrove interface. Fully automatic
-// generation is not yet possible due to user-generated types in the external
-// interface lost in the abi generation.
+// be kept up-to-date with the actual Mangrove interface.
 
 pragma solidity >=0.7.0 <0.9.0;
 
 import "mgv_src/core/MgvLib.sol";
 
+///@title Interface for the Mangrove contract.
 interface IMangrove is HasMgvEvents {
   // # Permit functions
 
+  ///@notice See {IERC20Permit-DOMAIN_SEPARATOR}.
+  ///@return the domain separator.
   function DOMAIN_SEPARATOR() external view returns (bytes32);
 
+  ///@notice See {IERC20Permit-PERMIT_TYPEHASH}.
+  ///@return The permit type hash.
   function PERMIT_TYPEHASH() external pure returns (bytes32);
 
+  ///@notice approves the spender to spend the amount of tokens on behalf of the caller.
```

Comment:
backticks: In general I am not a big fan unless its important. Reading prose is easier on the eyes.

amount=value: I am unsure about this permit api since it is borrowed from somewhere, maybe we should keep it as is?

The intend of this exercise was not to revisit APIs :D if we do, then I'm not the one to answer. @adhusson ?

Reply:
value is the standard name for this `approve` argument I would say let's stick with it https://eips.ethereum.org/EIPS/eip-20#approve